orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -38,17 +39,74 @@ def _ortvalue_from_torch_tensor(torch_tensor):
     return C.OrtValue.from_dlpack(to_dlpack(torch_tensor), is_bool_tensor)
 
 
-def _torch_tensor_from_dl_pack(dlpack, ortvalue, device):
-    torch_tensor = from_dlpack(dlpack) if device.type != 'ort' else C.ort_from_dlpack(dlpack)
-    return torch_tensor.to(torch.bool) if ortvalue.data_type() == 'tensor(bool)' else torch_tensor
+def _ortvalues_to_torch_tensor(ortvalues, device):
+    if len(ortvalues) == 0:
+        return tuple()
+    if (hasattr(ortvalues, 'to_dlpack') and
+            all(ortvalue.proto_type() != 9 for ortvalue in ortvalues)):
+        if 'ort' == (device if isinstance(device, str) else device.type):
+            return tuple(ortvalues.to_dlpack(
+                lambda dlpack_structure: C.OrtValue.from_dlpack(dlpack_structure, False)))
```

Comment:
I still don't quite understand why "C.OrtValue.from_dlpack" here. This function should returns torch tensor list, but C.OrtValue.from_dlpack return OrtValue. And the original code uses "C.ort_from_dlpack", which is different. There are one more place below using "C.OrtValue.from_dlpack", which will also return OrtValue list instead of torch tensor list.